CMakeLists.txt

```diff
@@ -161,11 +161,18 @@ target_link_libraries(test_thin_plate_spline ${PROJECT_NAME} ${catkin_LIBRARIES}
 # To build tests, install google's testing framework
 # https://github.com/google/googletest/
 
-IF(GTEST_FOUND)
-    include_directories(system ${GTEST_INCLUDE_DIRS})
-    add_executable(test_timings tests/timings_tests)
-    target_link_libraries(test_timings ${PROJECT_NAME} ${GTEST_LIBRARIES})
-ENDIF(GTEST_FOUND)
+
+#############
+## Testing ##
+#############
+
```

Comment:
I think you're supposed to wrap the `catkin_add_gtest` stuff in `if(CATKIN_ENABLE_TESTING)` in ROS. This does make it slightly more annoying to run tests on it though, as you need to use `catkin_make run_tests`.

Reply:
Hmm, I'm following the documentation here: http://wiki.ros.org/gtest
I thought `if(CATKIN_ENABLE_TESTING)` was relevant for `add_rostest`?

I also use catkin-tools, so `catkin build` instead of `catkin_make`. 
As it is currently, it behaves as expected using `catkin build`. Tests are not built nor run under just `catkin build`. https://catkin-tools.readthedocs.io/en/latest/verbs/catkin_build.html#building-and-running-tests

Reply:
Well, I can't find any real documentation about `CATKIN_ENABLE_TESTING`, so maybe it was just a `rostest` thing?

Reply:
If you ever find good documentation for this, please let me know